### PR TITLE
feat: Use unity catalog instead of hive metastore

### PR DESCRIPTION
**Pull request by ebbeknudsen**
## Description

Use unity catalog instead of hive metastore when getting data from views in databricks

- Use `CatalogName` appsetting when running in Azure (populated from infrastructure)
- Use `hive_metastore` as `CatalogName` when running tests

I've verified that the subsystem test queries on dev002 actually are querying the unity catalog instead of hive metastore:
image
`ctl_shres_d_we_002` is the name of the unity catalog on d002

## References

- [x] Required infrastructure PR: https://github.com/Energinet-DataHub/dh3-infrastructure/pull/1907
- [x] Required automations PR for d001: https://github.com/Energinet-DataHub/dh3-automation/pull/459
- [x] Required automations PR for d002: https://github.com/Energinet-DataHub/dh3-automation/pull/457
- Task: https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/252 => https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10524051443/job/29160300618

## Checklist
- ~~[ ] Should the change be behind a feature flag?~~
- [x] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [x] Subsystem test executed (dev_002/dev_003) => https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10524051443/job/29160300618
- [x] Is there time to monitor state of the release to Production?
- [x] Reference to the task => https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/252

### CODE REVIEW

File: source/B2BApi/local.settings.sample.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "WorkspaceToken": "<Databricks workspace token>",
     "WorkspaceUrl": "<Databricks workspace url>",
     "WarehouseId": "<Databricks warehouse id>",
+    "EdiDatabricks__CatalogName": "<Databricks unity catalog name>",
```
Comment (lasrinnil) [inline source/B2BApi/local.settings.sample.json:38]:
```suggestion
    "CatalogName": "<Databricks unity catalog name>",
```

Comment (lasrinnil) [reply source/B2BApi/local.settings.sample.json:38]: Why the "EDI" Databrikcks prefix?

**Review by lasrinnil — APPROVED** [review]
Great work! Please update the title to reflect this is only for Energy 👍

File: source/B2BApi/local.settings.sample.json (hunk 1 above)
Comment (ebbeknudsen) [reply source/B2BApi/local.settings.sample.json:38]: `EdiDatabricks` is the name of the options section: 
image

Which corresponds to an app-setting.json that looks something like this:
```json
{
    "EdiDatabricks": {
        "CatalogName": "xyz"
    }
}
```

**Pull request merged**